plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt

```diff
@@ -211,7 +219,7 @@ internal class OrtModelBuilder : ToolingModelBuilder {
                                 selectedComponent.dependencies.toOrtDependencies(poms, visited + id)
                             }
 
-                            OrtDependencyImpl(
+                            val dependency = OrtDependencyImpl(
```

Comment:
To avoid the need to introduce this variable, use a construct like

```
OrtDependencyImpl(
    ...
).also {
    ortDependencyCache[selectedComponent] = it
}
```

Similar below again.